apps/server/src/routes/chat.ts

```
@@ -1,112 +1,471 @@
+    await this.setupAuth(token);
   }
 
-  if (canSendMessages.data.unlimited) {
-    console.log('User has unlimited access');
-  } else if (!canSendMessages.data.balance) {
-    console.log('No balance and not unlimited');
-    return c.json({ error: 'Insufficient plan quota' }, 403);
-  } else if (canSendMessages.data.balance <= 0) {
-    console.log('Balance is 0 or less');
-    return c.json({ error: 'Insufficient plan balance' }, 403);
+  async onChatMessage(onFinish: StreamTextOnFinishCallback<{}>) {
```

Biome (1.9.4)

[error] Don't use '{}' as a type.

Prefer explicitly define the object shape. '{}' means "any non-nullable value".

(lint/complexity/noBannedTypes)